packages/react-core/src/core/providers/thirdweb-provider.tsx

```diff
@@ -114,14 +114,23 @@ export const ThirdwebProviderCore = <TChains extends Chain[]>({
 
   const activeChainWithKey = useMemo(() => {
     if (typeof activeChainIdOrObjWithKey === "number") {
-      return supportedChainsWithKey.find(
+      const resolveChain = supportedChainsWithKey.find(
         (chain) => chain.chainId === activeChainIdOrObjWithKey,
       );
+      if (!resolveChain) {
+        throw new Error(`Invalid chainId: ${activeChainIdOrObjWithKey}`);
+      }
+      return resolveChain;
     }
+
     if (typeof activeChainIdOrObjWithKey === "string") {
-      return supportedChainsWithKey.find(
+      const resolvedChain = supportedChainsWithKey.find(
         (chain) => chain.slug === activeChainIdOrObjWithKey,
       );
+      if (!resolvedChain) {
+        throw new Error(`Invalid chain: ${activeChainIdOrObjWithKey}`);
```

Comment:
Could we say something like: "The activeChain is not part of the supportedChains prop" so that it gives a better hint to devs?